src/fstar/FStar.Interactive.Ide.fs

```diff
@@ -144,26 +137,24 @@ let deps_of_our_file filename =
 
 (* .fsti name (optional) * .fst name * .fsti recorded timestamp (optional) * .fst recorded timestamp  *)
 type m_timestamps = list<(option<string> * string * option<time> * time)>
+type deps_stack_t = list<env_t> // Environments created while typechecking deps
 
 (*
  * type check remaining dependencies and record the timestamps.
- * m is the current module name, not the module name of the dependency. it's actually a dummy that is pushed on the stack and never used.
- * it is used for type checking the fragments of the current module, but for dependencies it is a dummy.
- * adding it as the stack entry needed it.
  * env is the environment in which next dependency should be type checked.
  * the returned timestamps are in the reverse order (i.e. final dependency first), it's the same order as the stack.
  * note that for dependencies, the stack and ts go together (i.e. their sizes are same)
- * returns the new stack, environment, and timestamps.
+ * returns the new environment, deps stack, and timestamps.
  *)
-let rec tc_deps (m:modul_t) (stack:stack_t)
-                (env:env_t) (remaining:list<string>) (ts:m_timestamps)
-//      : stack<'env,modul_t> * 'env * m_timestamps
+let rec tc_deps (deps_stack: deps_stack_t) (env: env_t)
+                (remaining: list<string>) (ts: m_timestamps)
   = match remaining with
-    | [] -> stack, env, ts
+    | [] -> env, (deps_stack, ts)
```

Comment:
can we pair the `deps_stack` with the `ts` (i.e. `List.combine` them) to enforce the invariant that they always have the same length?

Reply:
Oh yes we can ^^ I take this to mean you're burning to review [cpitclaudel_912](https://github.com/FStarLang/FStar/commits/cpitclaudel_912), which does that and more ? :)